app/src/processing/app/Base.java

```diff
@@ -1206,36 +1178,157 @@ public void rebuildExamplesMenu(JMenu menu) {
       menu.addSeparator();
     }
 
+    // Libraries can come from 4 locations: collect info about all four
+    File ideLibraryPath = BaseNoGui.getContentFile("libraries");
+    File sketchbookLibraryPath = BaseNoGui.getSketchbookLibrariesFolder();
+    File platformLibraryPath = null;
+    File referencedPlatformLibraryPath = null;
+    String platformName = null;
+    String referencedPlatformName = null;
+    String myArch = null;
+    TargetPlatform targetPlatform = BaseNoGui.getTargetPlatform();
+    if (targetPlatform != null) {
+      myArch = targetPlatform.getId();
+      platformName = targetPlatform.getPreferences().get("name");
+      platformLibraryPath = new File(targetPlatform.getFolder(), "libraries");
+      String core = BaseNoGui.getBoardPreferences().get("build.core", "arduino");
+      if (core.contains(":")) {
```

Comment:
If TargetPlatform has a function to get the referenced platform within its API, I agree it should be used rather than parsing the build.core pref.  Likewise for finding the human-readable name of the platform.  But I didn't see anything like this inside TargetPlatform.  I'd be happy to change this to use an official API... I just need someone to tell me which API does this?